core/rawdb/accessors_chain.go

```diff
@@ -329,21 +329,41 @@ func ReadHeaderRange(db ethdb.Reader, number uint64, count uint64) []rlp.RawValu
 	return rlpHeaders
 }
 
+const HashByHeader = false
```

Comment:
This definitely needs some docs, perhaps something like this: 
```suggestion
// HashByHeader is a compile-time constant, which can be used to toggle the behaviour of ReadHeaderRLP. If `false`,
// it is assumed that the header hash == `crypto.Keccak256(header_rlp)`. This may not be true in all ethereum-forks.
// Such forks need to set this constant to `true`, to force the hash-check to utilize the `header.Hash` method. 
const HashByHeader = false
```